### PR TITLE
New iso generation

### PR DESCRIPTION
**Pull request by ariscop**
## Purpose
Simplify iso and cab generation, using less code and better handling dependencies
Opening this pull request is to trigger intergration tests and get feedback, not mergable yet if only because of the changed output locations for iso and cab files

## Proposed changes
Features added in cmake 3.0, 3.1, and 3.2 make it possible to dynamically add dependencies to custom commands, allowing one to make custom targets (like the cab and iso targets in this pr) in a more general way, and avoids a 'generation' step like the one create_iso_lists implemented

### CODE REVIEW

File: boot/bootdata/packages/CMakeLists.txt [hunk 1]
```diff
@@ -30,7 +23,7 @@ add_custom_command(
     COMMAND ${CMAKE_COMMAND} -D SRC1=${CMAKE_CURRENT_SOURCE_DIR}/reactos.dff.in
                              -D SRC2=${CMAKE_CURRENT_BINARY_DIR}/reactos.dff.$<CONFIG>.dyn
                              -D DST=${CMAKE_CURRENT_BINARY_DIR}/reactos.dff.maydiff
-                             -P ${CMAKE_CURRENT_BINARY_DIR}/concat.cmake
+                             -P ${REACTOS_SOURCE_DIR}/sdk/tools/concat.cmake
```
Comment (HBelusca) [inline boot/bootdata/packages/CMakeLists.txt]: Should be in sdk/cmake/ instead.

File: CMakeLists.txt [hunk 2]
```diff
@@ -253,7 +255,8 @@ else()
     # For MSVC builds, this puts all debug symbols file in the same directory.
     set(CMAKE_PDB_OUTPUT_DIRECTORY "${REACTOS_BINARY_DIR}/msvc_pdb")
 
-    #begin with boot so reactos_cab target is defined before all other modules
+    #begin with dist so reactos_cab target is defined before all other modules
```
Comment (HBelusca) [inline CMakeLists.txt]: Space after #  please.

**Comment by HBelusca** [comment]
Note to self: to be put in regard with [Introduce the experimental all-in-one ReactOS BootCD (setup_improvements)](https://github.com/HBelusca/reactos/commit/59f47ff4e2bc6e65260744b7e52ba97ee3316a45)

**Comment by ariscop** [comment]
@HBelusca x86/x64 wouldn't be hard, i'm looking into running setupapi offline so one could build an image from cab files rather than integrating it into the build system as is done currently

**Comment by HBelusca** [comment]
>  running setupapi offline so one could build an image from cab files [...]

It looks you are basically redoing what cabman does already.

**Comment by ariscop** [comment]
It's not directly relevant to this pr which is about generalizing iso and cab files in cmake; i can explain better on irc how i think image generation could be improved

**Pull request closed without merging**